src/tests/multihost/alltests/test_ldap_password_policy.py

```diff
@@ -0,0 +1,236 @@
+        tools = sssdTools(multihost.client[0])
+        tools.sssd_conf("nss", {'filter_groups': 'root',
+                                'filter_users': 'root',
+                                'debug_level': '9'}, action='update')
+        ldap_params = {'use_fully_qualified_names': False}
+        tools.sssd_conf('domain/%s' % (ds_instance_name), ldap_params)
+        client_hostname = multihost.client[0].sys_hostname
+        # setup 389-ds w/server-side pw controls
+        cn_config = 'cn=config'
+        for element, value in [('passwordExp', [b'on']),
+                               ('passwordMaxAge', [b'86400']),
+                               ('passwordWarning', [b'86400'])]:
+            ldap_modify_ds(multihost, ldap.MOD_REPLACE, cn_config, element, value)
+        # set the password
+        user_dn = 'uid=ppuser1,ou=People,dc=example,dc=test'
+        ldap_modify_ds(multihost, ldap.MOD_REPLACE, user_dn, 'userPassword', [b'Secret123'])
+        client.run_command('> /var/log/secure')
+        client.run_command("systemctl stop sssd ;"
+                           "rm -rf /var/log/sssd/* ; "
+                           "rm -rf /var/lib/sss/db/* ; "
+                           "systemctl start sssd")
+        auth_success(client_hostname, 'ppuser1', 'Secret123')
+        time.sleep(3)
+        file_scure = '/var/log/secure'
+        file_ssd = f'/var/log/sssd/sssd_{ds_instance_name}.log'
+        find_logs(multihost, file_scure, "Your password will expire in ")
+        find_logs(multihost, file_ssd, "Server returned control [1.3.6.1.4.1.42.2.27.8.5.1]")
+        find_logs(multihost, file_ssd, "Password will expire in [86")
+        with pytest.raises(AssertionError):
+            find_logs(multihost, file_ssd,
+                      "Server does not support the requested "
+                      "control [1.3.6.1.4.1.42.2.27.8.5.1]")
+
+    @staticmethod
+    def test_maxage(multihost):
+        """
+        :title: "Set passwordMaxAge to 1"
+        :id: fc68293e-9676-11ed-9808-845cf3eff344
+        :steps:
+            1. setup 389-ds w/server-side passwordMaxAge to 1
+            2. Check user was to change password
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+        """
+        client = multihost.client[0]
+        client_hostname = multihost.client[0].sys_hostname
+        cn_config = 'cn=config'
+        ldap_modify_ds(multihost, ldap.MOD_REPLACE, cn_config, 'passwordMaxAge', [b'1'])
+        user_dn = 'uid=ppuser1,ou=People,dc=example,dc=test'
+        ldap_modify_ds(multihost, ldap.MOD_REPLACE, user_dn, 'userPassword', [b'Secret123'])
+        client.run_command('> /var/log/secure')
+        client.run_command("systemctl stop sssd ;"
+                           "rm -rf /var/log/sssd/* ; "
+                           "rm -rf /var/lib/sss/db/* ; "
+                           "systemctl start sssd")
+        file_location = '/script/password_expire.sh'
```

Comment:
Hve a look at this automation of password change if it will not work here:
https://github.com/SSSD/sssd/blob/master/src/tests/multihost/sssd/testlib/common/utils.py#L651